### PR TITLE
Issue1108 get time span tmy3

### LINKED ISSUE #1108: getTimeSpan (used by weather data reader) fails on some Windows installations
On some Windows installations (namely the JModelica Jenkins CI), the weather data reader fails to simulate. The simulation log is
```
FMIL: module = Model, log level = 3: [WARNING][FMU status:Warning]   <ModelicaError category="warning"><value name="msg">"Syntax error at character 1 of&#10;ICAL/EXTREME PERIODS,6,Summer - Week Nearest Max Temperature ...&#10;Syntax error at character 1 of&#10;ICAL/EX"</value></ModelicaError>
FMIL: module = Model, log level = 2: [ERROR][FMU status:Error]   <DependentParametersEvaluationFailed category="error">Error evaluating dependent parameters and start values</DependentParametersEvaluationFailed>
FMIL: module = Model, log level = 2: [INFO][FMU status:Error]   Error evaluating dependent parameters. Initialization failed.
Enter Initialize returned with an error. Enable logging for more information, (FMUModel(..., enable_logging=True)).
```
The error is caused by `IBPSA.BoundaryConditions.WeatherData.BaseClasses.getTimeSpanTMY3`.

It appears that correcting the line endings from Unix to Windows fixes this problem, but this would not be robust as users may provide weather files with Unix file endings.

We should consider rewriting `getTimeSpan` to simply use C with `fscanf` to do this file parsing.

@AnaConstantin : For this we should also add regression test which are missing for `getTimeSpanTMY3`.

### PR DESCRIPTION
**Pull request by mwetter**
This closes #1108

### CODE REVIEW

File: IBPSA/Resources/C-Sources/getTimeSpan.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ * ---------------------
+ * Get start and end time of weather data.
+ *
+ * filename: weather data file path
+ * tabNam: name of table on weather file
+ * timeSpan: vector [start time, end time]
+ */
+int getTimeSpan(const char * filename, const char * tabNam, double* timeSpan) {
+  double firstTimeStamp, lastTimeStamp, interval;
+  int rowCount, colonCount;
```
Comment (Mathadon) [inline IBPSA/Resources/C-Sources/getTimeSpan.c]: doesn't this have to be 'columnCount'?
Similarly, all other 'colon' variables have to renamed?

File: IBPSA/Resources/C-Sources/getTimeSpan.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+   }
+
+  lastColonIndicator = concat("#C", colonCountString);
+  free(colonCountString);
+
+  while (1) {
+    getline(&line, &len, fp);
+    rowIndex++;
+    if (strstr(line, lastColonIndicator)) {
+      break;
```
Comment (Mathadon) [inline IBPSA/Resources/C-Sources/getTimeSpan.c]: This implementation assumes that the weather file is correctly formatted. I would either be more conservative and simply skip all rows that start with '#', or keep the current implementation and verify that the next row does not start with '#' and throw an error if it does.

File: IBPSA/Resources/C-Sources/getTimeSpan.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  }
+  retVal = fscanf(fp, "%lf", &lastTimeStamp);
+  if (retVal == EOF){
+    ModelicaFormatError("Received unexpected EOF in getTimeSpan.c when searching last time stamp in %s.",
+    filename);
+  }
+  free(line);
+  fclose(fp);
+
+  /* find average time inteval */
```
Comment (Mathadon) [inline IBPSA/Resources/C-Sources/getTimeSpan.c]:
```suggestion
  /* find average time interval */
```

**Comment by mwetter** [comment]
@Mathadon : thanks for the changes.
@JayHuLBL : can you please address these.

**Comment by JayHuLBL** [comment]
@Mathadon Thanks for your comments. I will address them.

**Comment by mwetter** [comment]
@Mathadon : This is ready to merge.

**Pull request merged**